packages/aws-cdk/test/integ/cli/bootstrapping.integtest.ts

```diff
@@ -44,6 +87,52 @@ test('can bootstrap multiple toolkit stacks', async () => {
       { Key: 'Foo', Value: 'Bar' },
     ]);
   } finally {
-    await deleteStacks(bootstrapStackName1, bootstrapStackName2);
+    await deleteBootstrapStack(bootstrapStackName1);
+    await deleteBootstrapStack(bootstrapStackName2);
   }
-});
\ No newline at end of file
+});
+
+/**
+ * Delete bootstrap stack, cleaning S3 bucket if necessary
+ */
+async function deleteBootstrapStack(stackName: string) {
+  try {
+    const response = await cloudFormation('describeStacks', { StackName: stackName });
+
+    const bucketOutput = (response.Stacks?.[0].Outputs ?? []).find(o => o.OutputKey === 'BucketName');
+    if (bucketOutput?.OutputValue) {
+      await emptyBucket(bucketOutput?.OutputValue);
+    }
+
+  } catch (e) {
+    if (e.message.indexOf('does not exist') > -1) { return; }
```

Comment:
What's happening here? I don't understand this condition...

Reply:
Eat the error if the stack does not exist.

Reply:
I think we're missing an abstraction here. How about extracting a helper function `findStack(stackName: string): Stack | undefined`, instead of directly using CloudFormation's `describeStacks` API, that would contain this logic?